homeassistant/components/media_player/firetv.py

```diff
@@ -4,166 +4,144 @@
 
+REQUIREMENTS = ['firetv==1.0.7']
+
 _LOGGER = logging.getLogger(__name__)
 
 SUPPORT_FIRETV = SUPPORT_PAUSE | \
     SUPPORT_TURN_ON | SUPPORT_TURN_OFF | SUPPORT_PREVIOUS_TRACK | \
-    SUPPORT_NEXT_TRACK | SUPPORT_SELECT_SOURCE | SUPPORT_VOLUME_SET | \
-    SUPPORT_PLAY
+    SUPPORT_NEXT_TRACK | SUPPORT_SELECT_SOURCE | SUPPORT_STOP | \
+    SUPPORT_VOLUME_SET | SUPPORT_PLAY
+
+CONF_ADBKEY = 'adbkey'
+CONF_GET_SOURCE = 'get_source'
+CONF_GET_SOURCES = 'get_sources'
 
-DEFAULT_SSL = False
-DEFAULT_DEVICE = 'default'
-DEFAULT_HOST = 'localhost'
 DEFAULT_NAME = 'Amazon Fire TV'
-DEFAULT_PORT = 5556
-DEVICE_ACTION_URL = '{0}://{1}:{2}/devices/action/{3}/{4}'
-DEVICE_LIST_URL = '{0}://{1}:{2}/devices/list'
-DEVICE_STATE_URL = '{0}://{1}:{2}/devices/state/{3}'
-DEVICE_APPS_URL = '{0}://{1}:{2}/devices/{3}/apps/{4}'
+DEFAULT_PORT = 5555
+DEFAULT_GET_SOURCE = True
+DEFAULT_GET_SOURCES = True
+
+
+def has_adb_files(value):
+    """Check that ADB key files exist."""
+    priv_key = value
+    pub_key = '{}.pub'.format(value)
+    cv.isfile(pub_key)
+    return cv.isfile(priv_key)
+
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
-    vol.Optional(CONF_DEVICE, default=DEFAULT_DEVICE): cv.string,
-    vol.Optional(CONF_HOST, default=DEFAULT_HOST): cv.string,
+    vol.Required(CONF_HOST): cv.string,
     vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
     vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
-    vol.Optional(CONF_SSL, default=DEFAULT_SSL): cv.boolean,
+    vol.Optional(CONF_ADBKEY): has_adb_files,
+    vol.Optional(CONF_GET_SOURCE, default=DEFAULT_GET_SOURCE): cv.boolean,
+    vol.Optional(CONF_GET_SOURCES, default=DEFAULT_GET_SOURCES): cv.boolean
 })
 
-
-def setup_platform(hass, config, add_entities, discovery_info=None):
-    """Set up the FireTV platform."""
-    name = config.get(CONF_NAME)
-    ssl = config.get(CONF_SSL)
-    proto = 'https' if ssl else 'http'
-    host = config.get(CONF_HOST)
-    port = config.get(CONF_PORT)
-    device_id = config.get(CONF_DEVICE)
-
-    try:
-        response = requests.get(
-            DEVICE_LIST_URL.format(proto, host, port)).json()
-        if device_id in response[CONF_DEVICES].keys():
-            add_entities([FireTVDevice(proto, host, port, device_id, name)])
-            _LOGGER.info("Device %s accessible and ready for control",
-                         device_id)
-        else:
-            _LOGGER.warning("Device %s is not registered with firetv-server",
-                            device_id)
-    except requests.exceptions.RequestException:
-        _LOGGER.error("Could not connect to firetv-server at %s", host)
+PACKAGE_LAUNCHER = "com.amazon.tv.launcher"
+PACKAGE_SETTINGS = "com.amazon.tv.settings"
 
 
-class FireTV:
-    """The firetv-server client.
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the FireTV platform."""
+    from firetv import FireTV  # pylint: disable=no-name-in-module
```

Comment:
We shouldn't need to disable this.

Reply:
Fixed in https://github.com/home-assistant/home-assistant/pull/17767/commits/8ecbd1f9301368676d8ef896cfe85d8fd91b2c46